README.md

```diff
@@ -21,22 +21,53 @@ Config(
 )
 ```
 
-**Important: Make sure to set a `max_entries` in your anyrun config. Without that, the plugin will be VERY slow, since there exist over 16 thousand options to search through at the time of writing this.**  
+**Important: Make sure to set a `max_entries` in your anyrun config. Without that, the plugin will be VERY slow, since there exist over 16 thousand options to search through at the time of writing this.**
 **I have set mine to 10, since even that value is enough for it to go off-screen, so you shouldn't lose any data.**
 
 # Using this with NixOS?
 
-1. Use the [home-manager module](https://github.com/NixOS/nixpkgs/blob/nixos-unstable)
-2. Add the plugin to it:
-   ```nix
-     programs.anyrun.config.plugins = [ inputs.anyrun-nixos-options.packages.${pkgs.system}.default ];
-   ```
+Under flakes, the following instructions will apply:
+
+1. Use the [anyrun home-manager module](https://github.com/Kirottu/anyrun/blob/master/nix/hm-module.nix)
+2. Add the plugin to your anyrun plugins list
+
+```nix
+programs.anyrun.config = {
+    # ...
+    plugins = [
+        inputs.anyrun-nixos-options.packages.${pkgs.system}.default
+        # other plugins that you might have
+    ];
+    # ...
+};
+```
+
 3. Create a config file for it:
-   ```nix
-     programs.anyrun.extraConfigFiles."nixos-options.ron".text = ''
-       Config(
-         options_path: "${config.system.build.manual.optionsJSON}/share/doc/nixos/options.json"
-       )
-     '';
-   ```
-4. Done.
+
+```nix
+#                  ↓ make sure osConfig is in the argument set
+{inputs, pkgs, osConfig,  ...}: {
+    programs.anyrun.extraConfigFiles."nixos-options.ron".text = let
+        #               ↓ home-manager refers to the nixos configuration as osConfig
+        nixos-options = osConfig.system.build.manual.optionsJSON + "/share/doc/nixos/options.json";
+        # get the docs-json package from the home-manager flake
+        hm-options = inputs.home-manager.packages.${pkgs.system}.docs-json + "/share/doc/home-manager/options.json";
+        # merge your options
+        options = builtins.toJSON ["${nixos-options}"];
+        # or alternatively if you wish to read any other documentation options, such as home-manager
+        # options = builtins.toJSON ["${nixos-options}" "${nixos-options}" "${some-other-option}" /* ... */];
```

Comment:
i think you meant the second `nixos-options` to be `hm-options`

Reply:
what do you mean, I clearly didn't